src/Controller/FeedbackController.php

```diff
@@ -0,0 +1,429 @@
+     *     )
+     * )
+     * @SWG\Response(
+     *     response=412,
+     *     description="Precondition failed.",
+     *     @SWG\Schema(
+     *     @SWG\Property(property="code", type="integer", example=412),
+     *     @SWG\Property(property="message",
+     *     type="string", example="You were not involved or the job is not yet finished!"),
+     *     )
+     * )
+     * @SWG\Response(
+     *     response=404,
+     *     description="Not found",
+     *     @SWG\Schema(
+     *     @SWG\Property(property="code", type="integer", example=404),
+     *     @SWG\Property(property="message", type="string", example="Not found!"),
+     *     )
+     * )
+     * @SWG\Response(
+     *     response="400",
+     *     description="Bad Request"
+     * )
+     * @SWG\Response(
+     *     response="200",
+     *     description="Successfull operation!",
+     *     @SWG\Schema(
+     *     @SWG\Property(property="message", type="string", example="Feedback submitted successfully!"),
+     *     )
+     * )
+     * @param Activity $activity
+     * @param User $userTo
+     * @param Request $request
+     * @param ValidationErrorSerializer $validationErrorSerializer
+     * @return JsonResponse
+     * @throws ORMException
+     * @throws OptimisticLockException
+     */
+    public function addFeedback(
+        Activity $activity,
+        User $userTo,
+        Request $request,
+        ValidationErrorSerializer $validationErrorSerializer
+    ): JsonResponse {
+        $authenticatedUser = $this->getUser();
+
+        $rights = $this->accessRightsPolicy->canGiveFeedback($activity, $authenticatedUser, $userTo);
+
+        if ($rights === false) {
+            return new JsonResponse([
+                'code' => Response::HTTP_PRECONDITION_FAILED,
+                'message' => 'You were not involved or the job is not yet finished!'
+            ], Response::HTTP_PRECONDITION_FAILED);
+        }
+
+        $data = $request->getContent();
+
+        /** @var DeserializationContext $context */
+        $context = DeserializationContext::create()->setGroups(array('AddFeedback'));
+
+        $feedbackDTO = $this->serializer->deserialize(
+            $data,
+            FeedbackDTO::class,
+            'json',
+            $context
+        );
+
+        $errors = $this->validator->validate($feedbackDTO, null, ['AddFeedback']);
+
+        if (count($errors) > 0) {
+            return new JsonResponse(
+                [
+                    'code' => Response::HTTP_BAD_REQUEST,
+                    'message' => 'Bad Request',
+                    'errors' => $validationErrorSerializer->serialize($errors)
+                ],
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+
+        $newFeedback = $this->feedbackTransformer->addFeedback($feedbackDTO, $authenticatedUser, $activity, $userTo);
+        $this->feedbackRepository->save($newFeedback);
+
+        $userTo->setStars($this->feedbackRepository->getStarsSum($userTo)
```

Comment:
I don't see any other usages for `getStarsSum()` and `countFeedback()`. Why not create a single fnction which returns the final result?